tests/generate.py

```diff
@@ -810,9 +823,137 @@ def client_auth() -> None:
+            )
+        with open(os.path.join(output_dir, f"{name}.cert.der"), "wb") as f:
+            f.write(cert.public_bytes(encoding=Encoding.DER))
+
+    all_keys: list[tuple[str, ANY_PRIV_KEY, serialization.PrivateFormat]] = [
+        ("rsa_2048_pkcs8", _rsa_privkey(2048), PrivateFormat.PKCS8),
+        ("rsa_3072_pkcs8", _rsa_privkey(3072), PrivateFormat.PKCS8),
+        ("rsa_4096_pkcs8", _rsa_privkey(4096), PrivateFormat.PKCS8),
+        # Note: pyca cryptography calls PKCS1 the "TraditionalOpenSSL" format.
+        ("rsa_2048_pkcs1", _rsa_privkey(2048), PrivateFormat.TraditionalOpenSSL),
+        ("rsa_3072_pkcs1", _rsa_privkey(3072), PrivateFormat.TraditionalOpenSSL),
+        ("rsa_4096_pkcs1", _rsa_privkey(4096), PrivateFormat.TraditionalOpenSSL),
+        ("ec_p256_pkcs8", _ec_privkey(ec.SECP256R1()), PrivateFormat.PKCS8),
+        ("ec_p384_pkcs8", _ec_privkey(ec.SECP384R1()), PrivateFormat.PKCS8),
+        ("ed25519_pkcs8", _ed25519_privkey(), PrivateFormat.PKCS8),
+    ]
+
+    def _write_test(test_name: str) -> None:
+        ee_cert_path = os.path.join(output_dir, f"{test_name}.cert.der")
+        privkey_path = os.path.join(output_dir, f"{test_name}.key.der")
+
+        other_keys = [key_name for key_name, _, _ in all_keys if key_name != test_name]
+        other_keys_include_strs = []
+        other_keys_assert_strs = []
+        for key_name in other_keys:
+            key_path = os.path.join(output_dir, f"{key_name}.key.der")
+            other_keys_include_strs.append(
+                f'let {key_name} = include_bytes!("{key_path}");'
+            )
+            other_keys_assert_strs.append(
+                f"assert!(matches!(ee.verify_private_key({key_name}.as_ref()), Err(Error::CertPrivateKeyMismatch)));"
+            )
+
+        alloc_gate = (
+            '#[cfg(feature = "alloc")]'
+            if test_name.startswith("rsa") or test_name.endswith("sec1")
+            else ""
+        )
+        other_keys_include_str = "\n".join(other_keys_include_strs)
+        other_keys_assert_str = "\n".join(other_keys_assert_strs)
+        print(
+            """
+        %(alloc_gate)s
+        #[test]
+        fn %(test_name)s() {
+          let ee = include_bytes!("%(ee_cert_path)s");
+          let ee = EndEntityCert::try_from(ee.as_ref()).unwrap();
+          let privkey = include_bytes!("%(privkey_path)s");
+
+          %(other_keys_include_str)s
+
+          assert!(ee.verify_private_key(privkey.as_ref()).is_ok());
+          %(other_keys_assert_str)s
+        }
+        """
+            % locals(),
+            file=output,
+        )
+
+    with trim_top("ee_keyverify.rs") as output:
+        for name, privkey, serialization_format in all_keys:
+            _generate_testcert(name, privkey, serialization_format)
+            _write_test(name)
+
+        # pyca cryptography doesn't have support for serializing SEC1 format EC keys, so we
+        # pre-generate a key/cert with OpenSSL for this case:
```

Comment:
It might be my own inexperience with pyca cryptography but I wasn't able to find a way to easily marshal an EC private key in SEC 1 format. It looks like there's support for unmarshalling public keys in this format, but not marshalling private keys.

I thought an OK compromise was to generate these ~4 test files by hand with OpenSSL leaving a comment here to reproduce the work if needed.

Reply:
@alex got any hints on this?

Reply:
Pretty sure you want https://cryptography.io/en/latest/hazmat/primitives/asymmetric/serialization/#cryptography.hazmat.primitives.serialization.PrivateFormat.TraditionalOpenSSL as the serialization format to get SEC1. (The name is a bit stupid, I can't remember why)

Reply:
Ohhhhh! Yes that did the trick. Thank you :-)